src/solc_0.8/assetERC1155/AssetBaseERC1155.sol

```diff
@@ -0,0 +1,847 @@
+import "../common/interfaces/IERC1155TokenReceiver.sol";
+import "../common/Libraries/ObjectLib32.sol";
+import "../common/BaseWithStorage/WithSuperOperators.sol";
+import "../asset/libraries/ERC1155ERC721Helper.sol";
+
+// solhint-disable max-states-count
+abstract contract AssetBaseERC1155 is
+    WithSuperOperators,
+    IERC1155
+{
+    using Address for address;
+    using ObjectLib32 for ObjectLib32.Operations;
+    using ObjectLib32 for uint256;
+
+    bytes4 private constant ERC1155_IS_RECEIVER = 0x4e2312e0;
+    bytes4 private constant ERC1155_RECEIVED = 0xf23a6e61;
+    bytes4 private constant ERC1155_BATCH_RECEIVED = 0xbc197c81;
+
+    mapping(address => uint256) private _numNFTPerAddress; // erc721
+    mapping(uint256 => uint256) private _owners; // erc721
+    mapping(address => mapping(uint256 => uint256)) private _packedTokenBalance; // erc1155
+    mapping(address => mapping(address => bool)) private _operatorsForAll; // erc721 and erc1155
+    mapping(uint256 => address) private _erc721operators; // erc721
+    mapping(uint256 => bytes32) internal _metadataHash; // erc721 and erc1155
+    mapping(uint256 => bytes) internal _rarityPacks; // rarity configuration per packs (2 bits per Asset)
+    mapping(uint256 => uint32) private _nextCollectionIndex; // extraction
+
+    mapping(address => address) private _creatorship; // creatorship transfer
+
+    mapping(address => bool) private _bouncers; // the contracts allowed to mint
+
+    // @note : Deprecated.
+    mapping(address => bool) private _metaTransactionContracts;
+
+    address private _bouncerAdmin;
+
+    bool internal _init;
+
+    bytes4 internal constant ERC165ID = 0x01ffc9a7;
+
+    uint256 internal _initBits;
+    address internal _predicate; // used in place of polygon's `PREDICATE_ROLE`
+
+    uint8 internal _chainIndex; // modify this for l2
+    uint256 private constant CHAIN_INDEX_OFFSET_MULTIPLIER = uint256(2)**(256 - 160 - 1 - 32);
+    uint256 private constant CHAIN_INDEX_MASK = 0x00000000000000000000000000000000000000000000007F8000000000000000;
+
+    address internal _trustedForwarder;
+
+    uint256[20] private __gap;
+    // solhint-enable max-states-count
+
+    event BouncerAdminChanged(address oldBouncerAdmin, address newBouncerAdmin);
+    event Bouncer(address bouncer, bool enabled);
+    event MetaTransactionProcessor(address metaTransactionProcessor, bool enabled);
+    event CreatorshipTransfer(address indexed original, address indexed from, address indexed to);
+    event Extraction(uint256 indexed fromId, uint256 toId);
+    event AssetUpdate(uint256 indexed fromId, uint256 toId);
+
+    function init(
+        address trustedForwarder,
+        address admin,
+        address bouncerAdmin,
+        address predicate,
+        uint8 chainIndex
+    ) public {
+        // one-time init of bitfield's previous versions
+        _checkInit(0);
+        _admin = admin;
+        _bouncerAdmin = bouncerAdmin;
+        _predicate = predicate;
```

Comment:
We can do that, but it was probably just for clarity of purpose. And since we are anyway not touching the state variables I would prefer to use them the same way - predicate for minting on L1<->L2 transfers and bouncers for minting.